tests/bsim/bluetooth/samples/battery_service/src/central_test.c

```diff
@@ -0,0 +1,338 @@
+	}
+	return BT_GATT_ITER_CONTINUE;
+}
+
+/* Callback for handling Battery Level Read Response */
+static uint8_t battery_level_read_cb(struct bt_conn *conn, uint8_t err,
+				     struct bt_gatt_read_params *params, const void *data,
+				     uint16_t length)
+{
+	TEST_ASSERT(err == 0, "Failed to read Battery Level (err %d)", err);
+	if (data) {
+		LOG_DBG("[READ] BAS Battery Level: %d%%\n", *(const uint8_t *)data);
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+static unsigned char battery_level_status_indicate_cb(struct bt_conn *conn,
+						      struct bt_gatt_subscribe_params *params,
+						      const void *data, uint16_t length)
+{
+	static int ind_received;
+
+	if (!data) {
+		LOG_INF("bas level status indication disabled\n");
+	} else {
+		printk("[INDICATION]  BAS Battery Level Status: ");
+		for (int i = 0; i < length; i++) {
+			printk("%02x ", ((uint8_t *)data)[i]);
+		}
+		printk("\n");
+
+		if (ind_received++ > BAS_BLS_IND_RECEIVED_COUNT) {
+			SET_FLAG(indication_count_reached);
+		}
+	}
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t battery_level_status_notify_cb(struct bt_conn *conn,
+					      struct bt_gatt_subscribe_params *params,
+					      const void *data, uint16_t length)
+{
+	static int notify_count;
+
+	if (!data) {
+		LOG_INF("[UNSUBSCRIBED]\n");
+		params->value_handle = 0U;
+		return BT_GATT_ITER_STOP;
+	}
+
+	printk("[NOTIFICATION] BAS Battery Level Status: ");
+	for (int i = 0; i < length; i++) {
+		printk("%02x ", ((uint8_t *)data)[i]);
+	}
+	printk("\n");
+
+	if (notify_count++ > BAS_BLS_NTF_RECEIVED_COUNT) {
+		SET_FLAG(notification_count_reached);
+	}
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static void read_battery_level(const struct bt_gatt_attr *attr)
+{
+	/* Read the battery level after subscribing */
+	static struct bt_gatt_read_params read_params;
+
+	read_params.func = battery_level_read_cb;
+	read_params.handle_count = 1;
+	read_params.single.handle = bt_gatt_attr_get_handle(attr);
+	read_params.single.offset = 0;
+	bt_gatt_read(default_conn, &read_params);
+}
+
+static void subscribe_battery_level(const struct bt_gatt_attr *attr)
+{
+	int err;
+
+	battery_level_notify_params = (struct bt_gatt_subscribe_params){
+		/* In Zephyr, it is common practice for the CCC handle
+		 * to be positioned two handles after the characteristic handle.
+		 */
+		.ccc_handle = bt_gatt_attr_get_handle(attr) + 2,
+		.value_handle = bt_gatt_attr_value_handle(attr),
+		.value = BT_GATT_CCC_NOTIFY,
+		.notify = battery_level_notify_cb,
+	};
+
+	err = bt_gatt_subscribe(default_conn, &battery_level_notify_params);
+	if (err && err != -EALREADY) {
+		TEST_FAIL("Subscribe failed (err %d)\n", err);
+	} else {
+		LOG_DBG("Battery level [SUBSCRIBED]\n");
+	}
+	read_battery_level(attr);
+}
+
+static void subscribe_battery_level_status(const struct bt_gatt_attr *attr)
+{
+	int err;
+
+	if (get_device_nbr() < 3) { /* First two devices: Indications */
+		battery_level_status_sub_params = (struct bt_gatt_subscribe_params){
```

Comment:
Why two?

Reply:
This is to test indications for multiple connections

Reply:
But why?

Reply:
As discussed, i tried with 3 devices, 2 central one for indication and other for notification and 1 peripheral.
But the problem is that babble kit doesn't allows odd devices. it only syncs when there are even devices

code ref
_int bk_sync_init(void)
{
	uint device_number = get_device_nbr();
	uint peer_number = device_number ^ 1;_

Reply:
As discussed, made 2 central devices, one for indication and other for notification